### ISSUE TITLE
Move to better face recognition models

### ISSUE THREAD

**Issue by florianwgnr** [nextcloud/recognize#1085]
### Describe the feature you'd like to request

I have installed Recognize together with Memories and so far everything works: Recognize scans my images for faces and groups them together. Great!

Yesterday, just for testing, i set up an instance of [immich](https://github.com/immich-app/immich/tree/main/machine-learning) and loaded the same photo library i have stored in my Nextcloud. Immich currently uses the [buffalo_l](https://github.com/deepinsight/insightface/tree/master) model from [InsightFace](https://insightface.ai/) for facial recognition, and after scanning i was blown away by the accuracy: The model not only found way more pictures, it was also way better at assigning them to the right person. To give you some numbers: recognize found 706 pictures of me, immich with buffalo_l found 3780 pictures. All correctly assigned with sometimes mind-blowing accuracy.

Since the results were that good, i thought about suggesting replacing of the face recognition model, e.g. with buffalo_l or similar, to further promote Nextcloud as the main storage for photos. I am aware that this is not a one-click replacement and might require a lot of effort or additional work, but as this might also affect ongoing discussions, e.g. https://github.com/nextcloud/recognize/discussions/865 regarding undetected faces, i wanted to share my thoughts with you.

### Describe the solution you'd like

Move to an improved face recognition model

### Describe alternatives you've considered

--

**Comment by bonswouar** [comment]
A better face recognition model would really be needed, current one isn't doing so great imo.
It seems face detection is done thanks to https://github.com/justadudewhohacks/face-api.js , and there haven't been any new commit there for the last 4 years..

I guess a PR on face-api.js with new models would be doable, but as far as I understand it would need to be available for tensorflow-js (see [my comment ](https://github.com/nextcloud/recognize/issues/986#issuecomment-1987285491)on image detection models)
@florianwgnr [There](https://github.com/AIInAi/tf-insightface) [are](https://github.com/auroua/InsightFace_TF) [some](https://github.com/luckycallor/InsightFace-tensorflow) tensorflow implementations of InsightFace that you quoted, but unfortunately they're all much older than the release of the model buffalo_l (septembre 2021 I believe)

**Comment by tetebueno** [comment]
How about this fork that's being mentioned in project's issues?: https://github.com/vladmandic/face-api

**Comment by bonswouar** [comment]
> How about this fork that's being mentioned in project's issues?: https://github.com/vladmandic/face-api

Looking at [the models](https://github.com/vladmandic/face-api/tree/master/model) this projects uses (that are apparently based on [those](https://github.com/justadudewhohacks/face-api.js-models)), I doubt it's better than Recognize

**Comment by bonswouar** [comment]
> How about this fork that's being mentioned in project's issues?: https://github.com/vladmandic/face-api

It doesn't look like [the models](https://github.com/vladmandic/face-api/tree/master/model) have been updated much

**Comment by Handkep** [comment]
Is it feasible to create a wrapper for [buffalo_l](https://github.com/deepinsight/insightface/tree/master) to substitute the face_api directly?

**Comment by bonswouar** [comment]
> Is it feasible to create a wrapper for [buffalo_l](https://github.com/deepinsight/insightface/tree/master) to substitute the face_api directly?

I guess it could be feasible, although the preferred way would probably be to implement https://github.com/nextcloud/recognize/issues/73 to have docker-based external app, which could theoretically load any model then...

**Comment by cyrill-b** [comment]
I believe the recently published SDK [here](https://github.com/HyperInspire/InspireFace) might be helpful for this effort.

**Comment by krillin666** [comment]
Is there any update on this field? 

Thanks for the amazing work!

**Comment by lpryszcz** [comment]
hi, I'm also interested in this. one could actually run `immich-machine-learning` docker and only add code on nextcloud end to interact with it.

**Comment by wansti** [comment]
I'm seeing some racial bias with the current model, e.g. literally every Asian person in a set of photos being recognized as the same person which seems... problematic. So +1 for a better model/training data.

**Comment by lpryszcz** [comment]
since there was little progress I moved toward immich for photo management and I'm syncing photos across devices using External storage from Nextcloud. it works but it has certain limitations

**Comment by TheSpaceGod** [comment]
@lpryszcz I assume you did that via folder mounts and not webdav right? My nextcloud is using server side encryption so that won't work for me :(. I'm waiting for Immich to support cloud native external library support so all Immich file accesses go through the nextcloud server to decrypt the images.

**Comment by janschafrich** [comment]
I just switched to Immich for face recognition of my photos stored in my Nextcloud server. Immich identified more than twice as many images that contained other people or me as Nextcloud Recognize did. 

I installed Nextcloud manually on my server and now added Immich Docker container. In Immich, I access my Nextcloud files as an external library. Setup took less than 2 hours. So far works just as intended. You can even decide whether Immich should be able to delete images and edit metadata, or just have read-only access.

**Comment by nikbpetrov** [comment]
> I just switched to Immich for face recognition of my photos stored in my Nextcloud server. Immich identified more than twice as many images that contained other people or me as Nextcloud Recognize did.
> 
> I installed Nextcloud manually on my server and now added Immich Docker container. In Immich, I access my Nextcloud files as an external library. Setup took less than 2 hours. So far works just as intended. You can even decide whether Immich should be able to delete images and edit metadata, or just have read-only access.

I did a similar migration a while back now; I'd encourage you to just do a raw [Immich import rather than using External Library](https://www.reddit.com/r/immich/comments/1qtubmd/comment/o3lsyht/?context=3&utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button).

**Comment by marcelklehr** [comment]
Hi. Please refrain from posting off topic comments.
